packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -23,27 +25,43 @@ type DropdownMenuContextValue = {
   onOpenToggle(): void;
 };
 
+const SubmenuContext = React.createContext<boolean | undefined>(undefined);
+const InsideContentContext = React.createContext<boolean>(false);
 const [DropdownMenuProvider, useDropdownMenuContext] = createContext<DropdownMenuContextValue>(
   DROPDOWN_MENU_NAME
 );
 
 type DropdownMenuOwnProps = {
   open?: boolean;
-  defaultOpen?: boolean;
   onOpenChange?(open: boolean): void;
+  defaultOpen?: boolean;
+  dir?: Direction;
 };
 
 const DropdownMenu: React.FC<DropdownMenuOwnProps> = (props) => {
-  const { children, open: openProp, defaultOpen, onOpenChange } = props;
+  const parentSubmenuContext = React.useContext(SubmenuContext);
+  const isInsideContent = React.useContext(InsideContentContext);
+  const isRootMenu = parentSubmenuContext === undefined;
+
+  return (
+    <SubmenuContext.Provider value={!isRootMenu && isInsideContent}>
+      <DropdownMenuImpl {...props} />
+    </SubmenuContext.Provider>
+  );
+};
+
+const DropdownMenuImpl: React.FC<DropdownMenuOwnProps> = (props) => {
+  const { children, open: openProp, defaultOpen, onOpenChange, dir } = props;
+  const isSubmenu = React.useContext(SubmenuContext);
   const triggerRef = React.useRef<HTMLButtonElement>(null);
   const [open = false, setOpen] = useControllableState({
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: onOpenChange,
   });
-
+  const DropdownMenu = isSubmenu ? MenuPrimitive.Sub : MenuPrimitive.Root;
```

Comment:
Same with this one we usually tend to do the ternary inline and when there's a common child like that, that's the one we pull out instead in a const.